src/javascript/app_2/Stores/base_store.js

```diff
@@ -270,7 +270,7 @@ export default class BaseStore {
 
     @action.bound
     disposeSwitchAccount() {
-        this.switch_account_disposer();
+        this.switch_account_disposer = null;
```

Comment:
`disposeSwitchAccount` is only called when we want to intercept `client switching mechanism` and stop client switch. So only after we call it (to stop client switch), we set it to null. I believe you can disregard this file entirely. if you have any question regarding this piece, drop me a slack message.

Reply:
@kellybinary 
Regarding the [Mobx document](https://mobx.js.org/refguide/when.html), `when` returns a disposer function to cancel the autorunner permanently so the `this.switch_account_disposer` should be a function. 
If you faced a case that `this.switch_account_disposer` was not a function, probably, it was a race condition and we should fix it.
Would you plz let @easteregg  know how to produce it?

@easteregg regarding the name convention, `switch_account_disposer` should be in camelcase instead of snake case.